spec/request/target/delete_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'rails_helper'
+
+describe 'POST #delete target', type: :request do
+  let!(:user)           { create(:user) }
+  let!(:target)         { create(:target, user: user) }
+  let(:data)            { parsed_data }
+  let(:target_id)       { target.id }
+
+  subject do
+    delete api_v1_target_path(id: target_id), headers: auth_headers, as: :json
+  end
+
+  context 'when valid' do
+    context 'when signed in' do
+      let!(:auth_headers) { auth_user_headers }
+
+      it 'gets a sucessful answer' do
+        subject
+        expect(response).to be_successful
+      end
+
+      it 'does delete the target' do
+        expect { subject }.to change(user.targets.reload, :count).by(-1)
+      end
+    end
+  end
+
+  context 'when invalid' do
+    context 'when not signed in' do
+      let!(:auth_headers) { {} }
+      it 'gets a unauthorized error' do
+        subject
+        expect(response).to have_http_status(401)
+      end
+
+      it 'does not modify targets' do
+        expect { subject }.not_to change(user.targets.reload, :count)
+      end
+    end
+
+    context 'when invalid target id' do
+      let!(:auth_headers)    { auth_user_headers }
+      let!(:another_user)    { create(:user) }
```

Comment:
I don't think this is necessary. In the factory below, don't specify the user, and the factory will create another user.